src/hooks/usePromptConfirmOnNavigate.ts

```diff
@@ -43,12 +52,16 @@ const usePromptConfirmOnNavigate = ({
 
     navigator.push = (...args: Parameters<typeof push>) => {
       const promptConfirmOnNavigate = async (args: Parameters<typeof push>) => {
-        // Await for the isConfirmed to either return true or false, depending on the users input
-        const confirm = await isConfirmed({ title, description });
-        if (confirm !== false) {
-
-          
+        const argsProperties = args[0] as IArgs;
+        if (argsProperties && argsProperties.hash && projectSideBarItems.includes(argsProperties.hash)) {
```

Comment:
The regex would be used here. So it would be checked that argsProperties.pathname matches the regex which tells us that the user is not trying to navigate to a different url.

Reply:
I actually noticed that it's also possible to check if the hash exists as it is used to navigate to a specific spot on a page and if user tries to navigate to another page, the hash won't exist

Reply:
The regex solution would've been better than the original one but I guess that we can also go with checking the hash as then we wouldn't have any values that we should check or should remain the same

Reply:
Yep, that sounds even better 👍